packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts

```diff
@@ -160,17 +198,15 @@ export class BaseTypesVisitor<TRawConfig extends RawTypesConfig = RawTypesConfig
   }
 
   InterfaceTypeDefinition(node: InterfaceTypeDefinitionNode, key: number | string, parent: any): string {
-    const optionalTypename = this.config.nonOptionalTypename ? '__typename' : '__typename?';
```

Comment:
GraphQL interfaces do not have a `__typename`, only concrete types do. Other than being off-spec, it caused problems for TypeScript interfaces.

I realize this is a breaking change and the tests will need to be updated.

Reply:
You are right, thank you for pointing that out.